integration_tests/src/main/python/cast_test.py

```diff
@@ -167,3 +166,8 @@ def test_cast_long_to_decimal_overflow():
             f.col('a').cast(DecimalType(18, -1))),
         conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
 
+@pytest.mark.skip(not is_before_spark_311(), reason="Only in Spark 3.1.1+ do we have incompatibility with negative decimal scale")
+def test_cast_string_to_decimal(spark_tmp_path):
```

Comment:
Should this be named `test_cast_string_to_negative_scale_decimal` or something similar to note that this isn't just a generic decimal cast test?  Otherwise one would wonder why this isn't being tested across all Spark versions.

Reply:
Done